src/appSelector.ts

```diff
@@ -153,6 +236,7 @@ export function findAppInFolder(folderUri: vscode.Uri): App | undefined {
         found = false;
         break;
     }
+    variant = getAppVariants(folderStr.replace(/^file?:\/\//, ''), appName);
```

Comment:
What happens if you call `getAppVariants` if no app is found (we go through the `default` branch of the switch) ?
Also, you don't discriminate between C or Rust app before calling `getAppVariants`, what happens if it is called but it's a Rust app ?

Reply:
Correct, it is only valid for C apps.
Fixed